packages/realm-server/tests/realm-server-test.ts

```diff
@@ -2587,6 +2596,185 @@ module('Realm Server', function (hooks) {
+      let json = response.body;
+      assert.deepEqual(
+        json,
+        {
+          data: null,
+          included: null,
+        },
+        '/_user response is correct',
+      );
+    });
+
+    test('subscription is not found', async function (assert) {
+      let user = await insertUser(dbAdapter, 'user@test', 'cus_123');
+      let response = await request
+        .get(`/_user`)
+        .set('Accept', 'application/vnd.api+json')
+        .set(
+          'Authorization',
+          `Bearer ${createJWT(testRealm, 'user@test', ['read', 'write'])}`,
+        );
+      assert.strictEqual(response.status, 200, 'HTTP 200 status');
+      let json = response.body;
+      assert.deepEqual(
+        json,
+        {
+          data: {
+            type: 'user',
+            id: user.id,
+            attributes: {
+              matrixUserId: user.matrixUserId,
+              stripeCustomerId: user.stripeCustomerId,
+              creditsAvailableInPlanAllowance: null,
+              extraCreditsAvailableInBalance: null,
+            },
+            relationships: {
+              subscription: null,
+            },
+          },
+          included: null,
+        },
+        '/_user response is correct',
+      );
+    });
+
+    test('user subscibes to a plan and has extra credit', async function (assert) {
+      let user = await insertUser(dbAdapter, 'user@test', 'cus_123');
+      let plan = await insertPlan(
+        dbAdapter,
+        'Creator',
+        12,
+        2500,
+        'prod_creator',
+      );
+      let subscription = await insertSubscription(dbAdapter, {
+        user_id: user.id,
+        plan_id: plan.id,
+        started_at: 1,
+        status: 'active',
+        stripe_subscription_id: 'sub_1234567890',
+      });
+      let subscriptionCycle = await insertSubscriptionCycle(dbAdapter, {
+        subscriptionId: subscription.id,
+        periodStart: 1,
+        periodEnd: 2,
+      });
+      await addToCreditsLedger(dbAdapter, {
+        userId: user.id,
+        creditAmount: 100,
+        creditType: 'extra_credit',
+        subscriptionCycleId: subscriptionCycle.id,
+      });
+      let stripeInvoicePaymentSucceededEvent = {
+        id: 'evt_1234567890',
+        object: 'event',
+        type: 'invoice.payment_succeeded',
+        data: {
+          object: {
+            id: 'in_1234567890',
+            object: 'invoice',
+            amount_paid: 12, // creator plan
+            billing_reason: 'subscription_cycle',
+            period_end: 3,
+            period_start: 2,
+            subscription: 'sub_1234567890',
+            customer: 'cus_123',
+            lines: {
+              data: [
+                {
+                  price: { product: 'prod_creator' },
+                },
+              ],
+            },
+          },
+        },
+      } as StripeInvoicePaymentSucceededWebhookEvent;
+
+      await handlePaymentSucceeded(
```

Comment:
Why is this needed? You already set up your user and subscription state needed for this test. I think you shouldn't be setting up state for your test using stripe handlers in this type of test.

Reply:
You're right, what I need is adding credits for `plan_allowance`. I updated it.